### PR TITLE
add hasMonitor()

### PR DESCRIPTION
**Pull request by WalterBright**
Instead of hackish dependencies on whether a __monitor field exists or not, make it a trait.

### CODE REVIEW

**Comment by WalterBright** [comment]
@jacob-carlborg  need your advice on whether a ClassKind.objc has a __monitor field or not.

**Review by JinShil — APPROVED** [review]

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/dclass.d [hunk 1]
```diff
@@ -683,6 +683,14 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
         checkOverlappedFields();
     }
 
+    /**************
+     * Returns: true if there's a __monitor field
+     */
+    final bool hasMonitor()
+    {
+        return classKind == ClassKind.d || classKind == ClassKind.objc;
```
Comment (jacob-carlborg) [inline src/dmd/dclass.d]: Objective-C classes should not have a monitor field.

**Review by jacob-carlborg — CHANGES_REQUESTED** [review]

File: src/dmd/dclass.d (hunk 1 above)
Comment (jacob-carlborg) [reply src/dmd/dclass.d]: Hmm, but I guess it does.

Comment (WalterBright) [reply src/dmd/dclass.d]: ok

Comment (jacob-carlborg) [reply src/dmd/dclass.d]: I guess it safest to leave it like it is.

**Review by jacob-carlborg — APPROVED** [review]

**Pull request merged**